testing/agent-exporter/src/main/java/io/opentelemetry/javaagent/testing/IgnoredTestTypesConfigurer.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.testing;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.instrumentation.api.config.Config;
+import io.opentelemetry.javaagent.extension.ignore.IgnoredTypesBuilder;
+import io.opentelemetry.javaagent.extension.ignore.IgnoredTypesConfigurer;
+
+@AutoService(IgnoredTypesConfigurer.class)
+public class IgnoredTestTypesConfigurer implements IgnoredTypesConfigurer {
+
+  @Override
+  public void configure(Config config, IgnoredTypesBuilder builder) {
+    // we don't want to instrument auto-generated mocks
+    builder
+        .ignoreClass("org.mockito")
+        .ignoreClass("com.zaxxer.hikari.metrics.IMetricsTracker$MockitoMock$");
+  }
+}
```

Comment:
I think it's reasonable for this to be in the hikaricp module (non-testing artifact), especially since it should affect perf due to the underlying trie (as opposed to the old loop), and users may want to run their own tests using the javaagent.

(I'm also ok with it here)